workflows/methylseq.nf

```diff
@@ -0,0 +1,219 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    VALIDATE INPUTS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
+
+// Validate input parameters
+WorkflowMethylseq.initialise(params, log)
+
+// Check input path parameters to see if they exist
+def checkPathParamList = [ params.input, params.multiqc_config, params.fasta, params.known_splices ]
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+// Check mandatory parameters
+if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input samplesheet not specified!' }
+
+// Trimming / kit presets
+if(params.pbat){
+    params.clip_r1 = 9
+    params.clip_r2 = 9
+    params.three_prime_clip_r1 = 9
+    params.three_prime_clip_r2 = 9
+}
+else if( params.single_cell ){
+    params.clip_r1 = 6
+    params.clip_r2 = 6
+    params.three_prime_clip_r1 = 6
+    params.three_prime_clip_r2 = 6
+}
+else if( params.epignome ){
+    params.clip_r1 = 8
+    params.clip_r2 = 8
+    params.three_prime_clip_r1 = 8
+    params.three_prime_clip_r2 = 8
+}
+else if( params.accel || params.zymo ){
+    params.clip_r1 = 10
+    params.clip_r2 = 15
+    params.three_prime_clip_r1 = 10
+    params.three_prime_clip_r2 = 10
+}
+else if( params.cegx ){
+    params.clip_r1 = 6
+    params.clip_r2 = 6
+    params.three_prime_clip_r1 = 2
+    params.three_prime_clip_r2 = 2
+}
+else if( params.em_seq ){
+    params.maxins = 1000
+    params.clip_r1 = 8
+    params.clip_r2 = 8
+    params.three_prime_clip_r1 = 8
+    params.three_prime_clip_r2 = 8
+}
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    CONFIG FILES
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+ch_multiqc_config        = file("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
+ch_multiqc_custom_config = params.multiqc_config ? Channel.fromPath(params.multiqc_config) : Channel.empty()
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT LOCAL MODULES/SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// SUBWORKFLOW: Consisting of a mix of local and nf-core/modules
+//
+include { INPUT_CHECK } from '../subworkflows/local/input_check'
+
+if( params.aligner =~ /bismark/ ){
+    include { BISMARK as ALIGNER } from '../subworkflows/local/bismark'
+} else if ( params.aligner == 'bwameth' ){
+    include { BWAMETH as ALIGNER } from '../subworkflows/local/bwameth'
+}
```

Comment:
This is really clever, I haven't seen it done in any nf-core workflows before, so I'm not sure if it'll be transparent enough to anyone reading the workflow.